examples/nidcpower/measure-record.py

```diff
@@ -204,18 +203,15 @@ def _on_close(event):
 
 except grpc.RpcError as rpc_error:
     error_message = rpc_error.details()
+    for metadatum in rpc_error.trailing_metadata() or []:
+        if metadatum.key == "ni-error":
+            error_message += f"\nError status: {metadatum.value.decode('utf-8')}"
```

Comment:
Did you retest the examples after adding `decode`? 

Based on the `grpcio` Cython source, I would expect `value` to be a `str` unless `key` ends in "-bin".
https://github.com/grpc/grpc/blob/master/src/python/grpcio/grpc/_cython/_cygrpc/metadata.pyx.pxi#L61

Reply:
OK the code I came up with in the end behaves correctly, passes mypy, and is formatted in a sane readable fashion:
```
    for entry in rpc_error.trailing_metadata() or []:
        if entry.key == "ni-error":
            value = entry.value if isinstance(entry.value, str) else entry.value.decode("utf-8")
            error_message += f"\nError status: {value}"
```
FYI @DavidCurtiss @reckenro @danestull @tdunkle

Reply:
I think `isinstance` is the key: it tells mypy that this code can handle `str`.

Apparently `assert isinstance(entry.value, str)` or `cast(str, entry.value)` will also work:
https://mypy.readthedocs.io/en/stable/type_narrowing.html